### PR TITLE
Kraft support for Integration Tests

### PR DESCRIPTION
**Pull request by mannoopj**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/integration/kafka/api/ProducerFailureHandlingTest.scala [hunk 1]
```diff
@@ -211,16 +227,27 @@ class ProducerFailureHandlingTest extends KafkaServerTestHarness {
...
+
+    admin = TestUtils.createAdminClient(brokers, listenerName,
+      TestUtils.securityConfigs(Mode.CLIENT,
+        securityProtocol,
+        trustStoreFile,
+        "adminClient",
+        TestUtils.SslCertificateCn,
+        clientSaslProperties))
+    TestUtils.createOffsetsTopicWithAdmin(admin, brokers)
+    //TestUtils.createOffsetsTopic(zkClient, brokers)
```
Comment (ahuang98) [inline core/src/test/scala/integration/kafka/api/ProducerFailureHandlingTest.scala]: there's a `createOffsetsTopic` defined in `KafkaServerTestHarness` which takes care of creating and deleting an admin client

File: core/src/test/scala/kafka/tools/GetOffsetShellTest.scala [hunk 2]
```diff
@@ -55,49 +58,72 @@ class GetOffsetShellTest extends KafkaServerTestHarness with Logging {
...
+    admin = TestUtils.createAdminClient(brokers, listenerName,
+      TestUtils.securityConfigs(Mode.CLIENT,
+        securityProtocol,
+        trustStoreFile,
+        "adminClient",
+        TestUtils.SslCertificateCn,
+        clientSaslProperties))
+
+    //TestUtils.createOffsetsTopic(zkClient, servers)
+    TestUtils.createOffsetsTopicWithAdmin(admin, brokers)
```
Comment (ahuang98) [inline core/src/test/scala/kafka/tools/GetOffsetShellTest.scala]: same comment here

File: core/src/test/scala/unit/kafka/server/OffsetFetchRequestTest.scala [hunk 3]
```diff
@@ -72,15 +73,16 @@ class OffsetFetchRequestTest extends BaseRequestTest {
...
   }
 
-  @BeforeEach
+  /*@BeforeEach
   override def setUp(testInfo: TestInfo): Unit = {
     doSetup(testInfo, createOffsetsTopic = false)
 
     TestUtils.createOffsetsTopic(zkClient, servers)
-  }
+  }*/
```
Comment (ahuang98) [inline core/src/test/scala/unit/kafka/server/OffsetFetchRequestTest.scala]: don't forget to remove

File: core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala [hunk 4]
```diff
@@ -128,8 +130,9 @@ class RequestQuotaTest extends BaseRequestTest {
     finally super.tearDown()
   }
 
-  @Test
-  def testResponseThrottleTime(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testResponseThrottleTime(quorum: String): Unit = {
```
Comment (ahuang98) [inline core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala]: looks like @vveicc has converted this test file already (8/14)

**Review by ahuang98 — APPROVED** [review]
thanks for working on this! left a few comments but otherwise looks good

**Comment by divijvaidya** [comment]
cc: @dengziming as you might be interested in reviewing this

File: core/src/test/scala/integration/kafka/api/ProducerFailureHandlingTest.scala [hunk 5]
```diff
@@ -38,6 +40,7 @@ class ProducerFailureHandlingTest extends KafkaServerTestHarness {
 
   val numServers = 2
 
+
```
Comment (dengziming) [inline core/src/test/scala/integration/kafka/api/ProducerFailureHandlingTest.scala]: nit: extra line breaks.

**Review by dengziming — COMMENTED** [review]
@mannoopj Please rebase your code onto trunk and resolve conflicts to trigger CI, currently we can't see the test result. 
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14295/1/#showFailuresLink

**Comment by mannoopj** [comment]
cc: @dengziming

**Review by dengziming — APPROVED** [review]
LGTM.

**Comment by dengziming** [comment]
I find that there are many flaky tests in `FetchRequestTest.scala`, can you help investigating them @mannoopj .
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14295/4/#showFailuresLink
[Build / JDK 11 and Scala 2.13 / kafka.server.FetchRequestTest.testLastFetchedEpochValidation(String).quorum=kraft](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14295/4/testReport/junit/kafka.server/FetchRequestTest/Build___JDK_11_and_Scala_2_13___testLastFetchedEpochValidation_String__quorum_kraft/)
[Build / JDK 11 and Scala 2.13 / kafka.server.FetchRequestTest.testLastFetchedEpochValidationV12(String).quorum=kraft](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14295/4/testReport/junit/kafka.server/FetchRequestTest/Build___JDK_11_and_Scala_2_13___testLastFetchedEpochValidationV12_String__quorum_kraft/)
[Build / JDK 11 and Scala 2.13 / kafka.server.FetchRequestTest.testFetchWithPartitionsWithIdError(String).quorum=kraft](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14295/4/testReport/junit/kafka.server/FetchRequestTest/Build___JDK_11_and_Scala_2_13___testFetchWithPartitionsWithIdError_String__quorum_kraft_2/)
[Build / JDK 11 and Scala 2.13 / kafka.server.FetchRequestTest.testLastFetchedEpochValidation(String).quorum=kraft](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14295/4/testReport/junit/kafka.server/FetchRequestTest/Build___JDK_11_and_Scala_2_13___testLastFetchedEpochValidation_String__quorum_kraft_2/)
[Build / JDK 11 and Scala 2.13 / kafka.server.FetchRequestTest.testLastFetchedEpochValidationV12(String).quorum=kraft](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14295/4/testReport/junit/kafka.server/FetchRequestTest/Build___JDK_11_and_Scala_2_13___testLastFetchedEpochValidationV12_String__quorum_kraft_2/)

**Comment by dengziming** [comment]
I create https://issues.apache.org/jira/browse/KAFKA-15566 for these flaky tests

**Comment by jolshan** [comment]
Ah I also filed a JIRA for this. They are not flaky and failing on most builds. I don't think we should have merged this. 
https://issues.apache.org/jira/browse/KAFKA-15589. We should definitely disable these.

**Pull request merged**